gufe/protocols/protocoldag.py

```diff
@@ -355,6 +355,8 @@ def _from_dict(cls, dct: dict):
 
 def execute_DAG(protocoldag: ProtocolDAG, *,
                 shared: Optional[PathLike] = None,
+                scratch_basedir: Optional[PathLike] = None,
```

Comment:
just `scratch`, or change `shared` to be `shared_basedir` (or `shared_dir`)

Reply:
Adjusted this in our live session today. `shared` is expected to already be established by caller to be fed into this function, while `scratch_basedir` is needed to spawn scratch spaces for the `ProtocolUnit`s executed in the course of this function. These scratch spaces are cleared after each unit's execution.